src/charm.py

```diff
@@ -41,83 +65,51 @@ def __init__(self, *args):
 
     def _update_prometheus_jobs(self, _):
         """Setup Prometheus scrape configuration for external targets."""
-        self.unit.status = ActiveStatus()
-        if not self.unit.is_leader():
-            return
-
-        jobs = self._scrape_jobs()
-        if not jobs:
-            return
-
-        for relation in self.model.relations[self._prometheus_relation]:
-            relation.data[self.app]["scrape_jobs"] = json.dumps(jobs)
+        if self.unit.is_leader():
```

Comment:
Why does the existing flat code flow have to be changed to this nested code flow ?

Reply:
flattened.